pkg/operator/controller/status/controller.go

```diff
@@ -341,6 +342,32 @@ func checkDNSAvailable(dns *operatorv1.DNS) bool {
 	return false
 }
 
+// computeOperatorUpgradeableCondition computes the operator's current
+// Upgradeable status state by examining the Upgradeable status of the DNS
+// operator
+func computeOperatorUpgradeableCondition(dns *operatorv1.DNS) configv1.ClusterOperatorStatusCondition {
+	var upgradeable bool
+	for _, cond := range dns.Status.Conditions {
+		if cond.Type == operatorv1.OperatorStatusTypeUpgradeable && cond.Status == operatorv1.ConditionTrue {
+			upgradeable = true
+		}
+	}
+	if !upgradeable {
+		return configv1.ClusterOperatorStatusCondition{
+			Type:    configv1.OperatorUpgradeable,
+			Status:  configv1.ConditionFalse,
+			Reason:  "DNSNotUpgradeable",
+			Message: fmt.Sprintf("DNS %s is not upgradeable", dns.Name),
```

Comment:
It might be useful to include `cond.Message` from the DNS's status condition in the clusteroperator's status condition.